### PR TITLE
Unit tests for useQuery

### PR DESCRIPTION
**Pull request by jackdclark**
Part of #9 

Add test coverage for `useQuery`

```
 PASS  test/unit/useQuery.test.js
  useQuery
    ✓ calls useClientRequest with query (22ms)
    ✓ calls useClientRequest with options (2ms)
    ✓ returns initial state from useClientRequest & refetch (2ms)
    ✓ bypasses cache when refetch is called (1ms)
    ✓ sends the query on mount if no data & no error (2ms)
    ✓ does not send the same query on mount if data is already present (2ms)
    ✓ does not send the query on mount if there is an error (1ms)
    ✓ adds query to ssrPromises when in ssrMode if no data & no error (3ms)
    ✓ does not add query to ssrPromises when in ssrMode if there is already data (1ms)
    ✓ does not add query to ssrPromises when in ssrMode if there is an error (1ms)
    ✓ does not add query to ssrPromises when in ssrMode if ssr is overridden in options (2ms)
    ✓ does not send the same query twice (2ms)
    ✓ sends the query again if the variables change (5ms)
    ✓ sends another query if the query changes (2ms)
```

### CODE REVIEW

File: test/unit/useQuery.test.js [hunk 1]
```diff
@@ -1,5 +1,123 @@
...
+    expect(mockQueryReq).toHaveBeenCalledTimes(1);
+  });
+
+  it('does not send the same query on mount if data is already present', () => {
+    mockState.data = { some: 'data' };
+    testHook(() => useQuery(TEST_QUERY), { wrapper: Wrapper });
+    expect(mockQueryReq).not.toHaveBeenCalled();
+  });
+
+  it('does not send the query on mount if there is already data', () => {
```
Comment (bmullan91) [inline test/unit/useQuery.test.js]:
```suggestion
  it('does not send the query on mount if there is already an error', () => {
```

File: test/unit/useQuery.test.js [hunk 2]
```diff
@@ -1,5 +1,123 @@
...
+    mockState.error = true;
+    testHook(() => useQuery(TEST_QUERY), { wrapper: Wrapper });
+    expect(mockQueryReq).not.toHaveBeenCalled();
+  });
+
+  it('does not send the query on mount if there is an error', () => {
```
Comment (bmullan91) [inline test/unit/useQuery.test.js]: Ah this is the correct test - the one above is a duplicate.

Comment (jackdclark) [reply test/unit/useQuery.test.js]: Nice spot, removed this

**Comment by jackdclark** [comment]
@Joezo @bmullan91 thanks for the reviews. I've added the following three tests to cover the `useEffect` conditional firing
```
    ✓ does not send the same query twice (2ms)
    ✓ sends the query again if the variables change (2ms)
    ✓ sends another query if the query changes (2ms)
```

**Pull request merged**